### PR TITLE
feat(scalars): add the validations for int and bigInt

### PR DESCRIPTION
**Pull request by alejandrocabriales**
## Ticket
https://trello.com/c/S2RNIMLT/685-1-numbers-int-float-bigint-negativeint-positiveint-nonnegativeint-1nonpositiveint-negativefloat-positivefloat-nonnegativefloat-n

## Description
- Should allow setting **minValue:** number as a minimum allowed value.
- Should show the proper message when the minValue is not met.
- Should allow setting **maxValue:** number as a maximum allowed value.
- Should show the proper message when the maxValue has been exceeded.
-  Should allow setting **allowNegative:** Boolean.
- Should allow negative numbers if the allowNegative config is true.
- Should allow setting **isBigInt:** Boolean — Whether large numbers beyond Int are allowed.

### CODE REVIEW

File: packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      : Number(value) > 0
+        ? true
+        : "Value must be a positive value";
+  };
+
+export const validateIsBigInt =
+  (props: NumberFieldProps) => (value: string) => {
+    const isLargeNumber = Math.abs(Number(value)) > Number.MAX_SAFE_INTEGER;
+    return isLargeNumber && !props.isBigInt
+      ? "Value is too large for standard integer, set isBigInt to true"
```
Comment (YasielCabrera) [inline packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts]: You should omit this part: "set isBigInt to true". This is a message to be presented to the users no developers, and users can not change the code

File: packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  };
+
+export const validatePrecision =
+  (props: NumberFieldProps) => (value: string) => {
+    if (props.precision === 0) return true;
+    if (props.precision === undefined) return true;
+    const decimalPart = value.toString().split(".")[1];
+    return decimalPart && decimalPart.length <= (props.precision ?? 0)
+      ? true
+      : `Value must have ${props.precision} decimal places`;
```
Comment (YasielCabrera) [inline packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts]: If precision is 0 and receives a value with 1 decimal place, it should fail, right?

**Review by YasielCabrera — CHANGES_REQUESTED** [review]

File: packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts (hunk 1 above)
Comment (alejandrocabriales) [reply packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts]: You are right, done

File: packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts (hunk 2 above)
Comment (alejandrocabriales) [reply packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts]: good catch , done

**Review by YasielCabrera — APPROVED** [review]

**Pull request merged**